crates/datasources/src/excel/table.rs

```diff
@@ -33,32 +45,26 @@ pub async fn excel_streaming_table(
         Cow::Owned(first.clone())
     });
 
-    let working_sheet_range = match workbook.worksheet_range(&sheet) {
-        Ok(r) => Ok(r),
-        Err(_) => Err(ExcelError::Load {
-            msg: "Failed to open .xlsx file.".to_owned(),
-        }),
-    };
-
-    let range = working_sheet_range?;
+    let range = workbook
+        .worksheet_range(&sheet)
+        .map_err(|e| Into::<ExcelError>::into(e))?;
     let schema = excel::infer_schema(&range, has_header, infer_schema_length)?;
 
-    // todo: expand this handle multiple sheets
-    let mut streams = Vec::<Arc<(dyn PartitionStream + 'static)>>::with_capacity(1);
-
-    streams.push(Arc::new(ExcelSheetPartitionStream::new(
-        Arc::new(schema.clone()),
-        Box::pin(futures::stream::iter(vec![
-            xlsx_sheet_value_to_record_batch(range, has_header, infer_schema_length),
-        ])),
-    )));
+    // TODO: expand this handle multiple sheets
+    let stream: Vec<Arc<(dyn PartitionStream + 'static)>> =
```

Comment:
I don't think you need to specify the type in the vector declaration `Vec<_>` should work.